File: src/main/kotlin/com/sourcegraph/cody/agent/protocol/ClientInfo.kt
package com.sourcegraph.cody.agent.protocol

import com.sourcegraph.cody.agent.ExtensionConfiguration

data class ClientInfo(
    var version: String,
    var workspaceRootUri: String? = null,
    var extensionConfiguration: ExtensionConfiguration? = null,
    var capabilities: ClientCapabilities? = null,
) {
  val name = "JetBrains"
}

File: agent/bindings/csharp/ClientInfo.cs
using System.Text.Json.Serialization;

namespace Cody.Core.Agent.Protocol
{
  public class ClientInfo
  {
    [JsonProperty(PropertyName = "name")]
    public string Name { get; set; }
    [JsonProperty(PropertyName = "version")]
    public string Version { get; set; }
    [JsonProperty(PropertyName = "ideVersion")]
    public string IdeVersion { get; set; }
    [JsonProperty(PropertyName = "workspaceRootUri")]
    public string WorkspaceRootUri { get; set; }
    [JsonProperty(PropertyName = "globalStateDir")]
    public string GlobalStateDir { get; set; }
    [JsonProperty(PropertyName = "workspaceRootPath")]
    public string WorkspaceRootPath { get; set; }
    [JsonProperty(PropertyName = "extensionConfiguration")]
    public ExtensionConfiguration ExtensionConfiguration { get; set; }
    [JsonProperty(PropertyName = "capabilities")]
    public ClientCapabilities Capabilities { get; set; }
  }
}
